Cargo.lock

```diff
@@ -967,6 +961,15 @@ dependencies = [
  "either",
 ]
 
+[[package]]
+name = "itertools"
+version = "0.12.1"
```

Comment:
This change makes itertools fragmentation

Reply:
unfortunately I can't bump itertools any higher, because of some deprecation warnings, I'll send another pr to fix this later